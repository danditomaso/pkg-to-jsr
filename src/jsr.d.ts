/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

/**
 * A JSON representation of a JSR configuration file.
 */
export interface JSRConfigurationFileSchema {
  /**
   * The name of this JSR package. Must be scoped
   */
  name: string;
  /**
   * The version of this JSR package.
   */
  version: string;
  exports:
    | string
    | {
        /**
         * This interface was referenced by `undefined`'s JSON-Schema definition
         * via the `patternProperty` "^\.(/.*)?$".
         */
        [k: string]: string;
      };
  publish?: {
    /**
     * List of files, directories or globs that will be included in the published package.
     */
    include?: string[];
    /**
     * List of files, directories or globs that will be excluded from the published package.
     */
    exclude?: string[];
    [k: string]: unknown;
  };
  [k: string]: unknown;
}
