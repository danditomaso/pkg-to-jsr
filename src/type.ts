export type Exports = Record<string, string> | { '.': string };

export type JSR = {
	name: string;
	version: string;
	publish: {
		include: string[] | undefined;
		exclude: string[] | undefined;
	};
	exports: Exports | undefined;
};
