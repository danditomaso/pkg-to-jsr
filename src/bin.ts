import process from 'node:process';

import { dirname, resolve } from 'pathe';
import { consola } from 'consola';

import { findPackageJSON, genJsrFromPkg, readPkgJSON, writeJsr } from '.';

const pkgJSONPath = await findPackageJSON({ cwd: process.cwd() });

const pkgJSON = await readPkgJSON(pkgJSONPath);

const rootDir = dirname(pkgJSONPath);
const jsrPath = resolve(rootDir, 'jsr.json');

const jsr = genJsrFromPkg({ pkgJSON });
await writeJsr(jsrPath, jsr);

consola.success(`Generated ${jsrPath}`);
