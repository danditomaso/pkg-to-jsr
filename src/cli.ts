#! /usr/bin/env node

import process from 'node:process';
import fs from 'node:fs/promises';
import { defineCommand, runMain } from 'citty';

import { isAbsolute, resolve } from 'pathe';
import { consola } from 'consola';

import { description, name, version } from '../package.json';
import { findPackageJSON, genJsrFromPackageJson, readPkgJSON, writeJsr } from '.';

function resolveJsrPath(root: string) {
	return resolve(root, 'jsr.json');
}

const main = defineCommand({
	meta: { name, version, description },
	args: {
		root: {
			type: 'string',
			description: 'root directory including package.json',
		},
	},
	async run({ args: { root } }) {
		/** current working directory or maybe file */
		const cwd
			= isAbsolute(root)
				? root
				: root != null
					? resolve(process.cwd(), root)
					: process.cwd();

		/* check if root is a directory */
		if (!(await fs.lstat(cwd)).isDirectory()) {
			consola.error(`${root} is not a valid root directory`);
			process.exit(1);
		}

		const pkgJSONPath = await findPackageJSON({ cwd });
		const jsrPath = resolveJsrPath(cwd);

		const pkgJSON = await readPkgJSON(pkgJSONPath);
		const jsr = genJsrFromPackageJson({ pkgJSON });

		await writeJsr(jsrPath, jsr);
	},
	cleanup({ args: { root } }) {
		consola.success(`Generated ${resolveJsrPath(root)}`);
	},
});

await runMain(main);
