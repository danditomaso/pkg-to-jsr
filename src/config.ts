import typia from 'typia';
import { loadConfig as _loadConfig } from 'unconfig';
import type { PkgToJsrConfig } from './type';

/**
 * Define the configuration object
 * @param config - The configuration object
 * @returns The configuration object
 */
export function defineConfig(config: PkgToJsrConfig): PkgToJsrConfig {
	return config;
}

export async function loadConfig(): Promise<PkgToJsrConfig> {
	const { config } = await _loadConfig({
		sources: [
			{
				files: 'pkg.config',
			},
		],
	});

	/* check the configuration object */
	typia.assertGuard<PkgToJsrConfig>(config);

	return config;
}
