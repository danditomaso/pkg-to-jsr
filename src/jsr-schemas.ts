import { z } from 'zod/v4-mini';

// Generated Zod schemas from JSR JSON Schema
export const JSRScopedNameSchema = z.string().check(
	z.regex(/^@[a-z0-9\-_]+\/[a-z0-9\-_]+$/),
);

export const JSRExportsSchema = z.union([
	z.string(),
	z.record(z.string(), z.string()),
]);

export const JSRPublishSchema = z.object({
	include: z.optional(z.array(z.string())),
	exclude: z.optional(z.array(z.string())),
});

export const JSRConfigurationSchema = z.object({
	name: JSRScopedNameSchema,
	version: z.optional(z.string()),
	license: z.optional(z.string()),
	exports: JSRExportsSchema,
	publish: z.optional(JSRPublishSchema),
});

// Helper schemas for validation
export const StartWithExclamationSchema = z.string().check(
	z.regex(/^!/),
);
export const StringSchema = z.string();

// Package.json schema for validation (passthrough for now)
export const PackageJsonSchema = z.unknown();

// Type inference from schemas
export type JSRScopedName = z.infer<typeof JSRScopedNameSchema>;
export type JSRJson = z.infer<typeof JSRConfigurationSchema>;
export type PackageJson = {
	name?: string;
	author?: string | { name: string };
	files?: string[];
	exports?: string | Record<string, unknown>;
	version?: string;
	jsrName?: string;
	jsrInclude?: string[];
	jsrExclude?: string[];
};

// Validation helpers
export function isStartWithExclamation(value: unknown): value is `!${string}` {
	return StartWithExclamationSchema.safeParse(value).success;
}

export function isString(value: unknown): value is string {
	return StringSchema.safeParse(value).success;
}

export function isJSRScopedName(value: unknown): value is JSRScopedName {
	return JSRScopedNameSchema.safeParse(value).success;
}
