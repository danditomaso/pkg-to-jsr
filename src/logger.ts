import process from 'node:process';
import { consola, type ConsolaInstance } from 'consola';

import { name } from '../package.json';

export const logger: ConsolaInstance = consola.withTag(name);

/**
 * Throw an error and exit the process
 * @internal
 */
export function _throwError(message: string): never {
	logger.error(message);
	if (process.env.NODE_ENV === 'test') {
		throw new Error(message);
	}
	process.exit(1);
}

/**
 * Handle zod validation error
 */
/* eslint-disable ts/no-unsafe-assignment, ts/no-unsafe-member-access, ts/no-unsafe-call, ts/no-non-null-asserted-optional-chain */
export function _zodErrorHandler<T>(validation: any): { data: T } {
	if (validation?.success === false) {
		const errorMessage = validation?.error?.errors?.map((error: any) => {
			const { path, message, code } = error;
			return `${path.join('.')} is invalid: ${message} (${code})`;
		}).join('\n') ?? 'Unknown validation error';
		_throwError(`Invalid configuration: ${errorMessage}`);
	}

	return { data: validation?.data! };
}
/* eslint-enable ts/no-unsafe-assignment, ts/no-unsafe-member-access, ts/no-unsafe-call, ts/no-non-null-asserted-optional-chain */
