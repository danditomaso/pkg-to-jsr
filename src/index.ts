import fs from 'node:fs/promises';
import { styleText } from 'node:util';
import * as semver from '@std/semver';
import { findUp } from 'find-up-simple';
import terminalLink from 'terminal-link';
import { JSRConfigurationSchema, type JSRJson, type JSRScopedName } from './jsr';
import { isJSRScopedName, isStartWithExclamation, isString, type PackageJson } from './jsr-schemas';
import { _throwError, logger } from './logger';

type Exports = JSRJson['exports'];

function bold(str: string): string {
	return styleText('bold', str);
}

/**
 * Find package.json in the directory hierarchy
 * if not found, throw an error
 */
export async function findPackageJSON({ cwd }: { cwd: string }): Promise<string> {
	const path = await findUp('package.json', { cwd });
	if (!isString(path)) {
		_throwError(`Cannot find package.json at ${cwd}`);
	}
	return path;
}

/**
 * Get include field from package.json
 */
export async function readPkgJSON(pkgJSONPath: string): Promise<PackageJson> {
	const pkgJSON = await fs.readFile(pkgJSONPath, 'utf-8');
	const parsed = JSON.parse(pkgJSON) as PackageJson;
	// For now, we just validate that it's parseable JSON
	return parsed;
}

/**
 * Write JSR to file
 */
export async function writeJsr(jsrPath: string, jsr: JSRJson): Promise<void> {
	try {
		return await fs.writeFile(jsrPath, JSON.stringify(jsr, null, '\t'));
	}
	catch (e: unknown) {
		_throwError(`Failed to write JSR to ${jsrPath}: ${e?.toString()}`);
	}
}

/**
 * Get JSR name from package.json
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getName({
 *   "name": "package",
 *   "author": "author"
 *  })
 * )
 * .toBe('@author/package');
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getName({
 *   "name": "package",
 *   "author": {
 *     "name": "author",
 *     "email": "example@example.com"
 *   }
 *  })
 * )
 * .toBe('@author/package');
 *```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getName({
 *   "name": "package",
 *   "jsrName": "@author/package"
 *  })
 * )
 * .toBe('@author/package');
 * ```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  () => getName({
 *   "name": "package",
 *  })
 * )
 * .toThrowError();
 * ```
 * the function will throw an error because it is not scoped name
 *
 */
export function getName(pkgJSON: PackageJson): JSRScopedName {
	const { name, author } = pkgJSON;
	const jsrName = pkgJSON.jsrName;

	if (isJSRScopedName(jsrName)) {
		return jsrName;
	}

	if (!isString(jsrName) && isJSRScopedName(name)) {
		return name;
	}

	if (!isString(jsrName) && isString(name) && (isString(author) || isString(author?.name))) {
		const _author = isString(author) ? author : author.name;
		const jsrName = `@${_author}/${name}`;
		if (isJSRScopedName(jsrName)) {
			return jsrName;
		}
	}

	const errorMessages = [
		jsrName != null ? `${bold(`jsrName: ${String(jsrName)}`)} is not a valid scoped package name` : undefined,
		jsrName == null && name != null && author == null ? `${bold(`name: ${String(name)}`)} is not a valid scoped package name` : undefined,
		`On JSR, all packages are contained within a scope. See ${terminalLink('https://jsr.io/docs/scopes', 'https://jsr.io/docs/scopes')} for more information`,
		`To fix this issue, you can choose one of the following options:`,
		'1. add jsrName field to package.json',
		'2. use scoped package name (ex: @author/package)',
		'3. add name & author field to package.json ( ex: { "name": "package", "author": { "name": "author" } } will be converted to "@author/package")',
	] as const;

	_throwError(errorMessages.join('\n'));
}

/**
 * generate include for JSR from package.json
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getInclude({
 *   "files": ["src", "dist", "!node_modules"]
 *  })
 * )
 * .toEqual(["src", "dist"]);
 * ```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getInclude({
 *   "files": ["dist", "!node_modules"],
 *   "jsrInclude": ["src"]
 *  })
 * )
 * .toEqual(["src", "dist"]);
 * ```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getInclude({
 *   "files": ["src", "dist", "!node_modules"],
 *   "jsrInclude": ["src"],
 *  })
 * )
 * .toEqual(["src", "dist"]);
 * ```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getInclude({
 *   "files": ["src", "dist", "!node_modules"],
 *   "jsrInclude": ["src"],
 *   "jsrExclude": ["dist"]
 *  })
 * )
 * .toEqual(["src"]);
 * ```
 */
export function getInclude(pkgJSON: PackageJson): string[] | undefined {
	const {
		files = [],
		jsrInclude = [],
		jsrExclude = [],
	} = pkgJSON;

	const includeFromFiles = files.filter(file => !isStartWithExclamation(file));

	const includes = [...jsrInclude, ...includeFromFiles].filter(file => !jsrExclude.includes(file));

	if (includes.length === 0) {
		return;
	}

	return Array.from(new Set(includes));
}

/**
 * generate exclude for JSR from package.json
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getExclude({
 *   "files": ["src", "dist", "!node_modules"]
 *  })
 * )
 * .toEqual(["node_modules"]);
 * ```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getExclude({
 *   "files": ["src", "dist", "!node_modules"],
 *   "jsrExclude": ["dist"]
 *  })
 * )
 * .toEqual(["dist", "node_modules"]);
 * ```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getExclude({
 *   "files": ["src", "dist", "!node_modules"],
 *   "jsrExclude": ["dist"]
 *  })
 * )
 * .toEqual(["dist", "node_modules"]);
 * ```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getExclude({
 *   "files": ["src", "dist", "!node_modules"],
 *   "jsrInclude": ["src"],
 *   "jsrExclude": ["dist"]
 *  })
 * )
 * .toEqual(["dist", "node_modules"]);
 * ```
 *
 */
export function getExclude(pkgJSON: PackageJson): string[] | undefined {
	const {
		files = [],
		jsrInclude = [],
		jsrExclude = [],
	} = pkgJSON;

	const excludeFromFiles = files
		.filter(file => isStartWithExclamation(file))
		.map((file) => {
			if (file.startsWith('!')) {
				return file.slice(1);
			}
			return file;
		});

	const excludes = [...jsrExclude, ...excludeFromFiles].filter(file => !jsrInclude.includes(file));

	if (excludes.length === 0) {
		return;
	}

	return Array.from(new Set(excludes));
}

/**
 * generate exports filed for JSR from package.json
 * the import path should be string, { jsr: string }, { import: string }
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getExports({
 *   "exports": "./index.js"
 *  })
 * )
 * .toEqual({ ".": "./index.js" });
 * ```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getExports({
 *   "exports": {
 *     ".": "./index.js",
 *     "./sub": "./sub.js"
 *   }
 *  })
 * )
 * .toEqual(
 *  {
 *    ".": "./index.js",
 *    "./sub": "./sub.js"
 *  }
 * );
 * ```
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  getExports({
 *   "exports": {
 *     ".": {
 *       "jsr": "./src/index.ts",
 *       "import": "./dist/index.js",
 *       "types": "./dist/index.d.ts"
 *     },
 *     "./sub": {
 *       "jsr": "./src/sub.ts",
 *       "import": "./dist/sub.js",
 *       "types": "./dist/sub.d.ts"
 *	   }
 *    }
 *  })
 * )
 * .toEqual(
 *  {
 *    ".": "./src/index.ts",
 *    "./sub": "./src/sub.ts"
 *  }
 * );
 * ```
 *
 */
export function getExports(pkgJSON: PackageJson): Exports {
	const { exports } = pkgJSON;

	if (exports == null) {
		throw new Error('No exports field found in package.json');
	}

	if (isString(exports)) {
		return { '.': exports };
	}

	const _exports = {} as Record<string, string>;

	for (const [key, value] of Object.entries(exports)) {
		// eslint-disable-next-line ts/switch-exhaustiveness-check
		switch (true) {
			case isString(value):
				_exports[key] = value;
				break;
			case typeof value === 'object' && value != null && 'jsr' in value && isString(value.jsr):
				/* if jsr is defined, use it */
				_exports[key] = value.jsr;
				break;
			case typeof value === 'object' && value != null && 'import' in value && (isString(value.import) || (typeof value.import === 'object' && value.import != null && 'default' in value.import && isString(value.import.default))):
				/* if import is defined, use it */
				_exports[key] = isString(value.import) ? value.import : (value.import as { default: string }).default;
				break;
			default:
				logger.warn(`Export key ${key} is ignored because it is not a string or object`);
		}
	}

	if (Object.keys(_exports).length === 0) {
		throw new Error('No valid exports field found in package.json');
	}

	return _exports;
}

/**
 * generate JSR from package.json
 *
 * @example
 * ```ts @import.meta.vitest
 * expect(
 *  genJsrFromPackageJson({
 *   pkgJSON: {
 *     "name": "package",
 *     "author": "author",
 *     "version": "1.0.0",
 *     "files": ["src", "dist", "!node_modules"],
 *     "exports": {
 *       ".": "./src/index.ts",
 *       "./sub": "./src/sub.ts"
 *     }
 *   }
 * })
 * )
 * .toEqual(
 *  {
 *    "name": "@author/package",
 *    "version": "1.0.0",
 *    "publish": {
 *      "include": ["src", "dist"],
 *      "exclude": ["node_modules"]
 *    },
 *    "exports": {
 *      ".": "./src/index.ts",
 *      "./sub": "./src/sub.ts"
 *    }
 *  }
 * );
 * ```
 */
export function genJsrFromPackageJson({ pkgJSON }: { pkgJSON: PackageJson }): JSRJson {
	const { version } = pkgJSON;

	const include = getInclude(pkgJSON);
	const exclude = getExclude(pkgJSON);
	const jsr = {
		name: getName(pkgJSON),
		version: version as string,
		publish: include == null && exclude == null
			? undefined
			: {
					...((include?.length ?? 0) > 0 && { include }),
					...((exclude?.length ?? 0) > 0 && { exclude }),
				},
		exports: getExports(pkgJSON),
	} as const satisfies JSRJson;

	/* check the JSR object */
	const validation = JSRConfigurationSchema.safeParse(jsr);

	if (!validation.success) {
		/* eslint-disable ts/no-unsafe-assignment, ts/no-unsafe-member-access, ts/no-unsafe-call */
		const error = validation.error as any;
		const errorMessage = error?.errors?.map((err: any) => {
			const { path, message, code } = err;
			return `${path.join('.')} is invalid: ${message} (${code})`;
		}).join('\n') ?? 'Unknown validation error';
		/* eslint-enable ts/no-unsafe-assignment, ts/no-unsafe-member-access, ts/no-unsafe-call */
		_throwError(`Invalid configuration: ${errorMessage}`);
	}

	const data = validation.data;

	if (data.version != null && !semver.canParse(String(data.version))) {
		_throwError(`Invalid version: ${String(data.version)}`);
	}

	return data;
}
