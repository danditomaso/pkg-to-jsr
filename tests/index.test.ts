import { resolve } from 'pathe';
import { expect, it } from 'vitest';
import { findPackageJSON, genJsrFromPackageJson, readPkgJSON } from '../src';

it('basic', async () => {
	const DIR = resolve(__dirname, './basic/');
	const pkgJsonPath = await findPackageJSON({ cwd: DIR });
	const pkgJSON = await readPkgJSON(pkgJsonPath);
	const jsr = genJsrFromPackageJson({ pkgJSON });

	await expect(JSON.stringify(jsr, null, '\t')).toMatchFileSnapshot(`${DIR}/jsr.json`);
});

it('without files', async () => {
	const DIR = resolve(__dirname, './without_files/');
	const pkgJsonPath = await findPackageJSON({ cwd: DIR });
	const pkgJSON = await readPkgJSON(pkgJsonPath);
	const jsr = genJsrFromPackageJson({ pkgJSON });

	await expect(JSON.stringify(jsr, null, '\t')).toMatchFileSnapshot(`${DIR}/jsr.json`);
});

it('gen name', async () => {
	const DIR = resolve(__dirname, './gen_name/');
	const pkgJsonPath = await findPackageJSON({ cwd: DIR });
	const pkgJSON = await readPkgJSON(pkgJsonPath);
	const jsr = genJsrFromPackageJson({ pkgJSON });

	await expect(JSON.stringify(jsr, null, '\t')).toMatchFileSnapshot(`${DIR}/jsr.json`);
});

it('jsr name', async () => {
	const DIR = resolve(__dirname, './jsrName/');
	const pkgJsonPath = await findPackageJSON({ cwd: DIR });
	const pkgJSON = await readPkgJSON(pkgJsonPath);
	const jsr = genJsrFromPackageJson({ pkgJSON });

	await expect(JSON.stringify(jsr, null, '\t')).toMatchFileSnapshot(`${DIR}/jsr.json`);
});

it('error caused name', async () => {
	const DIR = resolve(__dirname, './error_caused_name_field/');
	const pkgJsonPath = await findPackageJSON({ cwd: DIR });
	const pkgJSON = await readPkgJSON(pkgJsonPath);
	expect(() => genJsrFromPackageJson({ pkgJSON })).toThrowError();
});
