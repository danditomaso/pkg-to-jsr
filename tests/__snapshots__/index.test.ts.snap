// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`exports exports_with_jsr 1`] = `
"{
	"name": "@ryoppippi/dummy",
	"version": "0.1.0",
	"exports": {
		".": "./src/index.ts"
	},
	"publish": {
		"include": [
			"dist"
		]
	}
}"
`;

exports[`exports exprts_without_jsr 1`] = `
"{
	"name": "@ryoppippi/dummy",
	"version": "0.1.0",
	"exports": {
		".": "./dist/index.js"
	},
	"publish": {
		"include": [
			"dist"
		]
	}
}"
`;

exports[`exports exports_without_jsr_imports_object 1`] = `
"{
	"name": "@ryoppippi/dummy",
	"version": "0.1.0",
	"exports": {
		".": "./dist/index.js"
	},
	"publish": {
		"include": [
			"dist"
		]
	}
}"
`;

exports[`files test without files 1`] = `
"{
	"name": "@dummy/dummy",
	"version": "0.1.0",
	"exports": {
		".": "./dist/index.js"
	}
}"
`;

exports[`files test includes 1`] = `
"{
	"name": "@ryoppippi/dummy",
	"version": "0.1.0",
	"exports": {
		".": "./dist/index.js"
	},
	"publish": {
		"include": [
			"src/**/*.ts",
			"dist"
		]
	}
}"
`;

exports[`files test excludes 1`] = `
"{
	"name": "@ryoppippi/dummy",
	"version": "0.1.0",
	"exports": {
		".": "./dist/index.js"
	},
	"publish": {
		"include": [
			"dist"
		],
		"exclude": [
			"src/**/*.ts",
			"node_modules"
		]
	}
}"
`;

exports[`name test gen name 1`] = `
"{
	"name": "@ryoppippi/dummy",
	"version": "0.1.0",
	"exports": {
		".": "./dist/index.js"
	}
}"
`;

exports[`name test jsr name 1`] = `
"{
	"name": "@ryoppippi/helloworld",
	"version": "0.1.0",
	"exports": {
		".": "./dist/index.js"
	}
}"
`;

exports[`basic 1`] = `
"{
	"name": "@ryoppippi/dummy",
	"version": "0.1.0",
	"exports": {
		".": "./dist/index.js"
	},
	"publish": {
		"include": [
			"dist"
		]
	}
}"
`;
